src/meta/src/controller/catalog/util.rs

```diff
@@ -417,8 +421,23 @@ impl CatalogController {
 
                     visit_stream_node_cont_mut(&mut pb_stream_node, |node| {
                         if let Some(NodeBody::Union(_)) = node.node_body {
-                            node.input.retain_mut(|input| {
-                                if let Some(NodeBody::Merge(merge_node)) = &mut input.node_body {
+                            node.input.retain_mut(|input| match &mut input.node_body {
```

Comment:
The `merge` branch of the upstream sink in the "create sink into table" process might be located within the `project`, which the current clean logic does not account for. This can result in a dirty table graph that cannot be recovered.

Reply:
Why specific to project? I'm thinking if Merge can be contained by other plan nodes as well.
Anyway this is a clear fix, just writing first, in case we need to fix more cases. cc @BugenZhao.

Reply:
cc @chenzl25

Reply:
That's by designed when introducing the `create sink into table`, Cc @shanicky for more context.

Reply:
It's for alter table column with sink upstream. See #17203 for more info.